ui-participant/src/hub/survey/SurveyView.test.tsx

```diff
@@ -78,14 +88,15 @@ describe('Renders a survey', () => {
   })
 
   it('autosaves question and page progress', async () => {
-    const { submitSpy } = setupSurveyTest(generateThreePageSurvey())
+    const { submitSpy, triggerAutosave } = setupSurveyTest(generateThreePageSurvey())
 
     await userEvent.click(screen.getByText('Green'))
     await userEvent.click(screen.getByText('Next'))
     expect(screen.getByText('You are on page2')).toBeInTheDocument()
     await userEvent.type(screen.getByText('text input'), 'my Text')
     await userEvent.click(screen.getByText('Next'))
-    await new Promise(r => setTimeout(r, 600))
+    triggerAutosave()
+    triggerAutosave()
```

Comment:
Why twice?

Reply:
good question, will add a comment.  This is testing what happens when autosaves fail, and how we handle resubmits, so we need to attempt to autosave twice